plugin/arcconf/arcconf.py

```diff
@@ -644,3 +644,187 @@ def volume_delete(self, volume, flags=0):
+        """
+
+        if not volume.plugin_data:
+            raise LsmError(
+                ErrorNumber.INVALID_ARGUMENT,
+                "Illegal input volume argument: missing plugin_data property")
+
+        volume_info = volume.plugin_data
+        ctrl_id = str(volume_info['ctrl_id'])
+        volume_id = str(volume_info['ld_id'])
+        volume_state = volume_info['ld_state']
+
+        try:
+            if volume_state == LOGICAL_DEVICE_OK:
+                raise LsmError(ErrorNumber.NO_STATE_CHANGE, 'Volume is already in Optimal state!')
+            else:
+                self._arcconf_exec(['SETSTATE', ctrl_id, 'LOGICALDRIVE', volume_id, 'OPTIMAL'], flag_force=True)
+        except ExecError:
+            volume_data = self._arcconf_exec(['GETCONFIGJSON', ctrl_id, 'LOGICALDRIVE', volume_id])
+            volume_json_data = self._filter_cmd_output(volume_data)['LogicalDrive']
+            volume_state = volume_json_data['state']
+            if volume_state == LOGICAL_DEVICE_OFFLINE:
+                raise LsmError(ErrorNumber.NO_STATE_CHANGE, 'Volume state has not changed!')
+
+            raise LsmError(ErrorNumber.PLUGIN_BUG, 'Volume-enable failed unexpectedly')
+        return None
+
+    @_handle_errors
+    def volume_ident_led_on(self, volume, flags=Client.FLAG_RSVD):
+        """
+
+        :param volume: volume id to be identified
+        :param flags: for future use
+        :return:
+        Depends on command:
+            arcconf identify <ctrlNo> logicaldrive <ldNo> time 3600
+
+            default led blink time is set to 1 hour
+        """
+        if not volume.plugin_data:
+            raise LsmError(
+                ErrorNumber.INVALID_ARGUMENT,
+                "Illegal input volume argument: missing plugin_data property")
+
+        volume_info = volume.plugin_data
+        ctrl_id = str(volume_info['ctrl_id'])
+        volume_id = str(volume_info['ld_id'])
+
+        try:
+            self._arcconf_exec(['IDENTIFY', ctrl_id, 'LOGICALDRIVE', volume_id, 'TIME', '3600'], flag_force=True)
+        except ExecError:
+            raise LsmError(ErrorNumber.PLUGIN_BUG, 'Volume-ident-led-on failed unexpectedly')
+        return None
+
+    @_handle_errors
+    def volume_ident_led_off(self, volume, flags=Client.FLAG_RSVD):
+        """
+
+        :param volume: volume id to stop identification
+        :param flags: for future use
+        :return:
+        Depends on command:
+            arcconf identify <ctrlNo> logicaldrive <ldNo> stop
+        """
+        if not volume.plugin_data:
+            raise LsmError(
+                ErrorNumber.INVALID_ARGUMENT,
+                "Illegal input volume argument: missing plugin_data property")
+
+        volume_info = volume.plugin_data
+        ctrl_id = str(volume_info['ctrl_id'])
+        volume_id = str(volume_info['ld_id'])
+
+        try:
+            self._arcconf_exec(['IDENTIFY', ctrl_id, 'LOGICALDRIVE', volume_id, 'STOP'], flag_force=True)
```

Comment:
The text width should be less than 80.

Reply:
done - commit id - feb56754a6df51c945c854845a18460800378573